### PR TITLE
Update stackdriver e2e testing

### PR DESCRIPTION
**Pull request by bskaplan**
Updates for #317 

## Proposed Changes
- Updates the infrastructure for stackdriver tests to better work with the current implementation
- add a test that verifies stackdriver metrics for the broker. The test is currently disabled because the stackdriver implementation is currently broken due to an upstream config change.

### CODE REVIEW

File: test/e2e/lib/broker.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/knative-gcp/test/e2e/lib/metrics"
+	pkgmetrics "knative.dev/pkg/metrics"
+)
+
+func AssertBrokerMetrics(t *testing.T, client *Client) {
+	sleepTime := 4 * time.Minute
```
Comment (grantr) [inline test/e2e/lib/broker.go:30]: This is a long time! Where did this value come from?

Comment (bskaplan) [reply test/e2e/lib/broker.go:30]: Observations that data would show up about 2 minutes after the request was processed and wanting a buffer to make sure the tests won't be flaky. I don't like it, but it makes the test pass.

Comment (grantr) [reply test/e2e/lib/broker.go:30]: Thanks for figuring out how to make this pass!

**Pull request merged**